File: Python/Iniciante/1045.py
# Problema 1045 - Beecrowd - Iniciante - Nível 4

# Lendo os três valores do usuário com a função "input()", separando
# os valores pelo espaço, e ordenando os valores de modo decrescente
# com a função "sorted()".
lados = input().split()

ladoA, ladoB, ladoC = sorted(map(float, lados), reverse=True)

# Cálculo dos valores e mosta a saída do exercício.
if ladoA >= (ladoB + ladoC):
    print("NAO FORMA TRIANGULO")
else:
    if (ladoA * ladoA) == ((ladoB * ladoB) + (ladoC * ladoC)):
        print("TRIANGULO RETANGULO")
    elif (ladoA * ladoA) > ((ladoB * ladoB) + (ladoC * ladoC)):
        print("TRIANGULO OBTUSANGULO")
    elif (ladoA * ladoA) < ((ladoB * ladoB) + (ladoC * ladoC)):
        print("TRIANGULO ACUTANGULO")

    # Ajusta a lista com os valores dos lados e com a propriedade
    # ".count" podemos contar quantos lados possuem o valor passado,
    # assim testamos se há lados iguais.
    lados = [ladoA, ladoB, ladoC]

    if lados.count(ladoA) == 2 or lados.count(ladoB) == 2:
        print("TRIANGULO ISOSCELES")
    if lados.count(ladoA) == 3:
        print("TRIANGULO EQUILATERO")

File: Rust/Iniciante/1045.rs
use std::io;
use std::error::Error;

fn main() -> Result<(), Box<dyn Error>> {
	let stdin = io::stdin();
	let mut buffer = String::new();

	stdin.read_line(&mut buffer)?;
	let tokens = buffer.split_whitespace();

	let mut numbers: Vec<f64> = tokens.map(|v| v.parse().unwrap()).collect();
	numbers.sort_by(|a, b| b.partial_cmp(a).unwrap());

	let a: f64 = numbers[0];
	let b: f64 = numbers[1];
	let c: f64 = numbers[2];

	if a >= (b + c) {
		println!("NAO FORMA TRIANGULO");
	} else if a.powf(2.0) == (b.powf(2.0) + c.powf(2.0)) {
		println!("TRIANGULO RETANGULO");
	} else if a.powf(2.0) > (b.powf(2.0) + c.powf(2.0)) {
		println!("TRIANGULO OBTUSANGULO");

		if (a == b && a != c) || (a == c && a != b) || (b == c && b != a) {
			println!("TRIANGULO ISOSCELES");
		}
	} else if a.powf(2.0) < (b.powf(2.0) + c.powf(2.0)) {
		println!("TRIANGULO ACUTANGULO");

		if a == b && a != c || a == c && a != b || b == c && b != c {
			println!("TRIANGULO ISOSCELES");
		}

		if a == b && b == c {
			println!("TRIANGULO EQUILATERO");
		}
	}

	Ok(())
}